### ISSUE TITLE
Flags and logos are not displayed

### ISSUE THREAD

**Issue by Jayky** [furlongm/openvpn-monitor#287]
Hi,

first of all: Thanks a lot for this great tool. I*m using it on an armbian with apache2 installed. The openvpn-monitor works but no logos nor flags are displayed. I've read about those issues (like #36) and non of the mentioned fixes worked for me.

I used for installation /var/www/html/openvpn-monitor, my logo is in the subdir images. I even tried to understand how the paths for the images are set in the .py script but my skills were not good enough.... What settings/configs or else should I send you, so that you could analyse what is not working?

**Comment by ynad** [comment]
> Hi,
> 
> first of all: Thanks a lot for this great tool. I*m using it on an armbian with apache2 installed. The openvpn-monitor works but no logos nor flags are displayed. I've read about those issues (like #36) and non of the mentioned fixes worked for me.
> 
> I used for installation /var/www/html/openvpn-monitor, my logo is in the subdir images. I even tried to understand how the paths for the images are set in the .py script but my skills were not good enough.... What settings/configs or else should I send you, so that you could analyse what is not working?

Logo img can either be a direct http link to an image, or a file located in `images` subdir. In this case, the config directive `logo=logo.jpg` must include only the filename, or an eventual sub-path under `images` folder.

See code around this line:
https://github.com/furlongm/openvpn-monitor/blob/79a2019de14fc46bb2108da9532485bcd23976e7/openvpn-monitor.py#L631

Flags aswell are stored locally under `images/flags/` subdir. Check if you have them. For me they are displayed correctly.

**Comment by Jayky** [comment]
Hi,

yes and that is the reason why I am looking for help here. I have my logo in `images` subdir. Also I checked `images/flags/` subdir and it is filled with *.png files. My geoip DB works AFAIK due to a position is shown on the map, when a device is connected. Any further ideas? Just it case it could help, here are some system information:
![grafik](https://github.com/user-attachments/assets/37c06e8f-f3fb-4ce5-8332-db55777e4a13)

**Comment by ynad** [comment]
> Hi,
> 
> yes and that is the reason why I am looking for help here. I have my logo in `images` subdir. Also I checked `images/flags/` subdir and it is filled with *.png files. My geoip DB works AFAIK due to a position is shown on the map, when a device is connected. Any further ideas? Just it case it could help, here are some system information: ![grafik](https://private-user-images.githubusercontent.com/11135819/388122707-37c06e8f-f3fb-4ce5-8332-db55777e4a13.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzIxODM3OTksIm5iZiI6MTczMjE4MzQ5OSwicGF0aCI6Ii8xMTEzNTgxOS8zODgxMjI3MDctMzdjMDZlOGYtZjNmYi00Y2U1LTgzMzItZGI1NTc3N2U0YTEzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTIxVDEwMDQ1OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNhYTc4OTM3OTc4YmVlZTQ2NTA5NjczMjM0OGQ3NzA1NjYyZmFkZGYzMmQzYmYzNDljMTg4NjRiZmJiOTUxYWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.C5eddiQTOvFdQ7kxGMOCv0eaV1dCk1wSUDOPrpKhp88)

Do files under `images` dir have correct file permissions? If it can't read them, you should see a warning log message on syslog or apache server logs.

I would try to manually run `openvpn-monitor.py` script in debug mode (if you use venv source it before):
`python3 openvpn-monitor.py -d`

It should print on stdout the HTML code result. Look for `<img` tags, check if `src` is pointing to the correct path. For instance mine is like this:
`<img alt="Logo" style="max-height:46px; padding-top:3px;" src="images/logo.png">`
Also note that logo should have have `max-height:46px`

Manually browse to the logo img, to see if you can see it in a browser:
 `http://myvpnserver/images/logo.png`

The same for flags, in a browser navigate for example to:
 `http://myvpnserver/images/flags/gb.png`

**Comment by Jayky** [comment]
Hi,

so here are my results:
` python3 /var/www/html/openvpn-monitor/openvpn-monitor.py -d

WARNING: Config file does not exist or is unreadable: ./openvpn-monitor.conf
INFO: Using config file: /etc/openvpn-monitor.conf
DEBUG:
 === begin section
{'site': 'Wolken-VPN', 'logo': 'grey.png', 'latitude': '50.367', 'longitude': '8.75', 'maps': 'True', 'maps_height': '500', 'geoip_data': '/mnt/intern/files/GeoLite2-City_20241008/GeoLite2-City.mmdb', 'datetime_format': '%d/%m/%Y %H:%M:%S'}
=== end section
[...]
<a class="navbar-brand" href="#">
Wolken-VPN OpenVPN Status Monitor</a>
</div><div class="collapse navbar-collapse" id="myNavbar">
<ul class="nav navbar-nav"><li class="dropdown">
<a class="dropdown-toggle" data-toggle="dropdown" href="#">VPN
<span class="caret"></span></a>
<ul class="dropdown-menu">
<li><a href="#wolken_vpn">Wolken VPN</a></li>
</ul></li>
<li><a href="#map_canvas">Map View</a></li>
</ul>
<a href="#" class="pull-right"><img alt="Logo" 
style="max-height:46px; padding-top:3px;" 
src="images/grey.png"></a>
</div></div></nav>
[...]
`
So yes, I've got a problem that is stated in the initial WARNING. What is mentioned there is strange, due to I have a config file in that directory. I had to solve that by creating a link in `/etc` that is pointing to that config file. But till now I have not found the reason for that. 
Furthermore the image is not visible by a browser when I tried `http://192.168.2.10/images/grey.png`. The whole directory is owned by user `www-data`

**Comment by furlongm** [comment]
Let me know if this is still an issue with v2

**Comment by Jayky** [comment]
Merry Christmas and thank you very much for the new version.

I noticed that the new version has a different folder structure as well as new directories for images (and flags) folder. Would the old config file still work with the new structure?

**Comment by Jayky** [comment]
Dear @furlongm ,

I was not able to run the new version. What I've done:

- `git clone https://github.com/furlongm/openvpn-monitor.git /var/www/html/openvpn-monitor-v2`
- Then I copied the existing config to /var/www/html/openvpn-monitor-v2/etc/openvpn-monitor/ and my icon pictures to /var/www/html/openvpn-monitor-v2/openvpn_monitor/static/images/
- Within the config file I changed maps=True to enable_maps=True.
- After that I changed the name of the path to /var/www/html/openvpn-monitor as it was for the previous version.

What is missing? Maybe my mistake will help to extend the README section.

**Comment by furlongm** [comment]
Are there any errors? Did you update the apache config and run the pip commands to install dependencies?

**Comment by Jayky** [comment]
Dear @furlongm ,

sorry, you were right, I forgot about those steps. By doing them, I had to learn a lot ;-)
```
sudo apt-get install python3.11-venv build-essential python3-dev
curl -fsSL https://deb.nodesource.com/setup_20.x | sudo bash -
sudo apt-get install -y nodejs
curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | sudo apt-key add -
echo "deb https://dl.yarnpkg.com/debian/ stable main" | sudo tee /etc/apt/sources.list.d/yarn.list
sudo apt update && sudo apt install yarn

cd /var/www/html/openvpn-monitor/
sudo yarnpkg --prod --modules-folder openvpn_monitor/static/dist install
sudo python3 -m venv .venv
sudo chmod a+x .venv/bin/activate
sudo chmod -R a+w .venv

source .venv/bin/activate
pip install -r requirements.txt
flask --app openvpn_monitor/app run --debug
deactivate
```
After using flask for the 1st time, I realized some important things about the new places for the openvpn-monitor.conf: 

- /etc/open-monitor/openvpn-monitor.conf
- /var/www/html/openvpn-monitor/openvpn_monitor/openvpn-monitor.conf

I solved it by linking to my file.
```
sudo ln -s /var/www/html/openvpn-monitor/etc/openvpn-monitor/openvpn-monitor.conf /etc/open-monitor/openvpn-monitor.conf
sudo ln -s /var/www/html/openvpn-monitor/etc/openvpn-monitor/openvpn-monitor.conf /var/www/html/openvpn-monitor/openvpn_monitor/openvpn-monitor.conf
```
Unfortunatly, I still have the following result:
```
$ source .venv/bin/activate
(.venv) $ flask --app openvpn_monitor/app run --debug
[2024-12-29 12:09:18,495] [5469] [ERROR] Config file does not exist or is unreadable: /etc/openvpn-monitor/openvpn-monitor.conf
[2024-12-29 12:09:18,495] [5469] [WARNING] Trying /var/www/html/openvpn-monitor/openvpn_monitor/openvpn-monitor.conf as config file
[2024-12-29 12:09:18,497] [5469] [INFO] Using loaded config:
[2024-12-29 12:09:18,497] [5469] [INFO] {'VPN1': {'host': 'localhost',
[2024-12-29 12:09:18,497] [5469] [INFO]           'name': 'Wolken VPN',
[2024-12-29 12:09:18,498] [5469] [INFO]           'port': 'nnnn',
[2024-12-29 12:09:18,498] [5469] [INFO]           'show_disconnect': 'False'},
[2024-12-29 12:09:18,498] [5469] [INFO]  'openvpn-monitor': {'datetime_format': '%d/%m/%Y %H:%M:%S',
[2024-12-29 12:09:18,498] [5469] [INFO]                      'enable_maps': 'True',
[2024-12-29 12:09:18,498] [5469] [INFO]                      'geoip_data': '/mnt/intern/files/GeoLite2-City_20241008/GeoLite2-City.mmdb',
[2024-12-29 12:09:18,498] [5469] [INFO]                      'latitude': '50.367',
[2024-12-29 12:09:18,498] [5469] [INFO]                      'logo': 'grey.png',
[2024-12-29 12:09:18,498] [5469] [INFO]                      'longitude': '8.75',
[2024-12-29 12:09:18,498] [5469] [INFO]                      'maps_height': '500',
[2024-12-29 12:09:18,499] [5469] [INFO]                      'site': 'Wolken-VPN'}}
[2024-12-29 12:09:18,499] [5469] [DEBUG] === begin section
{'site': 'Wolken-VPN', 'logo': 'grey.png', 'latitude': '50.367', 'longitude': '8.75', 'enable_maps': 'True', 'maps_height': '500', 'geoip_data': '/mnt/intern/files/GeoLite2-City_20241008/GeoLite2-City.mmdb', 'datetime_format': '%d/%m/%Y %H:%M:%S'}
=== end section
[2024-12-29 12:09:18,499] [5469] [DEBUG] === begin section
OrderedDict([('VPN1', {'host': 'localhost', 'port': 'nnnn', 'name': 'Wolken VPN', 'show_disconnect': False})])
=== end section
 * Serving Flask app 'openvpn_monitor/app'
 * Debug mode: on
[2024-12-29 12:09:18,533] [5469] [INFO] WARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.
 * Running on http://127.0.0.1:5000
[2024-12-29 12:09:18,533] [5469] [INFO] Press CTRL+C to quit
[2024-12-29 12:09:18,535] [5469] [INFO]  * Restarting with stat
Traceback (most recent call last):
  File "/var/www/html/openvpn-monitor/.venv/lib/python3.11/site-packages/flask/cli.py", line 961, in run_command
    app: WSGIApplication = info.load_app()  # pyright: ignore
                           ^^^^^^^^^^^^^^^
  File "/var/www/html/openvpn-monitor/.venv/lib/python3.11/site-packages/flask/cli.py", line 349, in load_app
    app = locate_app(import_name, name)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/www/html/openvpn-monitor/.venv/lib/python3.11/site-packages/flask/cli.py", line 255, in locate_app
    raise NoAppException(f"Could not import {module_name!r}.") from None
flask.cli.NoAppException: Could not import 'app'.
 * Debugger is active!
 * Debugger PIN:
```
 Now, I have no more idea how to continue...

**Comment by Jayky** [comment]
For the sake of completeness, here is a comment: I also looked into whether I could solve this with the configuration in Apache2, but unfortunately, I was unsuccessful. Therefore, I cannot offer a solution for those who have a similar issue.
I have now resolved it for myself by reverting to the previous version. I didn't delete it, I just renamed it. Renaming the paths again restored the old working state, even though there are no logos and flags. 
Nevertheless, thank you @furlongm for developing this tool.
